tests/FunctionRejectTest.php

```diff
@@ -21,7 +21,7 @@ public function shouldRejectAnException()
 
     /**
      * @test
-     * @expectedException \InvalidArgumentException
+     * @expectedException \TypeError
```

Comment:
Are those tests testing the type declarations still required at all? For example, we also do not have such tests for the `callable` declarations of `then()`. Not sure what's common/best practice...

Reply:
Mainly left them there because the test show intended behaviour, if someone comes in and wants to change that the test will give a heads up about that.

Reply:
I agree that this is not *required* as passing an invalid argument is not really handled by this library. I think that it might make sense in a higher-level test to show how this bubbles through the promise handling logic (see `shouldThrowWhenCalledWithANonException`). I'm okay with it either way :+1: